mackerel-plugin-gcp-compute-engine/lib/gcp-compute-engine.go

```diff
@@ -0,0 +1,202 @@
+			{Name: "dropped_bytes_count", Label: "Dropped Bytes Count"},
+		},
+	},
+	"Firewall.DroppedPacketsCount": mp.Graphs{
+		Label: "FireWall Dropped Packets Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "dropped_packets_count", Label: "Dropped Packets Count"},
+		},
+	},
+	"Cpu.Utilization": mp.Graphs{
+		Label: "CPU Utilization",
+		Unit:  "percentage",
+		Metrics: []mp.Metrics{
+			{Name: "utilization", Label: "Utilization"},
+		},
+	},
+	"Disk.BytesCount": mp.Graphs{
+		Label: "Disk Read Bytes Count",
+		Unit:  "bytes",
+		Metrics: []mp.Metrics{
+			{Name: "read_bytes_count", Label: "Read Bytes Count"},
+			{Name: "write_bytes_count", Label: "Write Bytes Count"},
+		},
+	},
+	"Disk.OpsCount": mp.Graphs{
+		Label: "Disk Read Ops Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "read_ops_count", Label: "Read Ops Count"},
+			{Name: "write_ops_count", Label: "Write Ops Count"},
+		},
+	},
+	"Network.BytesCount": mp.Graphs{
+		Label: "Network Received Bytes Count",
+		Unit:  "bytes",
+		Metrics: []mp.Metrics{
+			{Name: "received_bytes_count", Label: "Received Bytes Count"},
+			{Name: "sent_bytes_count", Label: "Sent Bytes Count"},
+		},
+	},
+	"Network.PacketsCount": mp.Graphs{
+		Label: "Network Received Packets Count",
+		Unit:  "float",
+		Metrics: []mp.Metrics{
+			{Name: "received_packets_count", Label: "Received Packets Count"},
+			{Name: "sent_packets_count", Label: "Sent Packets Count"},
+		},
+	},
+}
+
+// GraphDefinition is return graphdef
+func (p ComputeEnginePlugin) GraphDefinition() map[string]mp.Graphs {
+	return graphdef
+}
+
+func getLatestValue(listCall *monitoring.ProjectsTimeSeriesListCall, filter string, startTime string, endTime string, opts *Option) (float64, error) {
+	res, err := listCall.Filter(filter).IntervalEndTime(endTime).IntervalStartTime(startTime).Do(*opts)
+
+	if err != nil || res == nil {
+		return 0, err
+	}
+
+	valuePtr := res.TimeSeries[0].Points[0].Value
+
+	var value float64
+	if valuePtr.Int64Value != nil {
+		value = float64(*valuePtr.Int64Value)
+	} else if valuePtr.DoubleValue != nil {
+		value = *valuePtr.DoubleValue
+	}
+
+	return value, nil
+}
+
+func mkFilter(domain string, metricName string, instance string) string {
+	filter := `metric.type = "` + domain + metricName + `"`
+	switch domain {
+	case computeDomain:
+		filter += " AND metric.label.instance_name = " + instance
+	}
+
+	return filter
+}
+
+// FetchMetrics fetches metrics from Google Monitoring API
+func (p ComputeEnginePlugin) FetchMetrics() (map[string]float64, error) {
+	now := time.Now()
+	formattedEnd := now.Format(zuluFormat)
+	m, _ := time.ParseDuration(duration)
+	formattedStart := now.Add(-m).Format(zuluFormat)
+	listCall := p.MonitoringService.Projects.TimeSeries.List(p.Project)
+
+	stat := map[string]float64{}
+	for _, metricName := range []string{
+		"/firewall/dropped_bytes_count",
+		"/firewall/dropped_packets_count",
+		"/instance/cpu/utilization",
+		"/instance/disk/read_bytes_count",
```

Comment:
According to [reference](https://cloud.google.com/monitoring/api/metrics#gcp-compute), disk-related metrics also have `device_name` label.
Do those disk-related metrics work only with `label.instance_name` specified?  (it's happy if they work as summed value over all devices in the instance.)